src/edfi-paging-test/edfi_paging_test/helpers/argparser.py

```diff
@@ -0,0 +1,120 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from configargparse import ArgParser  # type: ignore
+from typing import List
+from dataclasses import dataclass
+
+
+@dataclass
+class MainArguments:
+    """
+    Container for holding arguments parsed at the command line.
+    """
+
+    baseUrl: str
+    connectionLimit: int
+    key: str
+    secret: str
+    output: str
+    contentType: str
+    retries: str
+    resourceList: List[str]
+    pageSize: int = 100
+
+
+def parse_main_arguments(args_in: List[str]) -> MainArguments:
+    """
+    Configures the command-line interface.
+    Parameters
+    ----------
+    args_in : list of str
+        Full argument list from the command line.
+    Returns
+    -------
+    arguments  : MainArguments
+        A populated `MainArguments` object.
+    """
+
+    parser = ArgParser()
+
+    parser.add(  # type: ignore
+        "-b",
+        "--baseUrl",
+        help="The base url used to derive the api, metadata, oauth, and dependency urls (e.g., http://server).",
+        type=str,
+        required=True,
+    )
+    parser.add(  # type: ignore
+        "-k",
+        "--key",
+        help="The web API OAuth key",
+        type=str,
+        required=True,
+    )
+    parser.add(  # type: ignore
+        "-s",
+        "--secret",
+        help="The web API OAuth secret",
+        type=str,
+        required=True,
+    )
+    parser.add(  # type: ignore
+        "-c",
+        "--connectionLimit",
+        help="Maximum concurrent connections to api",
+        type=int,
+        default=4,
+    )
+    parser.add(  # type: ignore
+        "-o",
+        "--output",
+        help="Directory for writing results",
+        type=str,
+        default="out",
+    )
+    parser.add(  # type: ignore
+        "-t",
+        "--contentType",
+        help="CSV or JSON",
+        type=str,
```

Comment:
```suggestion
        type=str,
        choices=["JSON", "CSV"]
```

LMS DS Loader [has an example of this](https://github.com/Ed-Fi-Alliance-OSS/LMS-Toolkit/blob/64abf05174c1e163cae3fa8d16dbd53fb8df7842/src/lms-ds-loader/edfi_lms_ds_loader/helpers/argparser.py#L125).

Works like an Enum.